server/tournaments/index.ts

```diff
@@ -485,10 +494,17 @@ export class Tournament extends Rooms.RoomGame<TournamentPlayer> {
 			output.errorReply(`${replacementUser.name} is already in the tournament.`);
 			return;
 		}
-		if (Tournament.checkBanned(this.room, replacementUser) || Punishments.isBattleBanned(replacementUser)) {
+		if (Tournament.checkBanned(this.room, replacementUser) || Punishments.isBattleBanned(replacementUser) ||
+		    replacementUser.namelocked) {
 			output.errorReply(`${replacementUser.name} is banned from joining tournaments.`);
 			return;
 		}
+		if ((this.room.settings.tournaments?.autoconfirmedOnly || this.autoconfirmedOnly) &&
+		   !user.autoconfirmed && !user.trusted) {
```

Comment:
```suggestion
		   !user.autoconfirmed) {
```
Trusted implicitly grants autoconfim.